src/Products.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Netzstrategen\ShopStandards;
+
+/**
+ * Meant to handle special Product features.
+ */
+class Products {
+
+  const POST_TYPE_PRODUCT = 'product';
+
+  const TAX_PRODUCT_CAT = 'product_cat';
+
+  const FIELD_ENFORCE_CAT_LINKS = Plugin::PREFIX . '_enforce_main_category_links';
+
+  /**
+   * Stores the product base link.
+   *
+   * @var string
+   */
+  private string $product_base_link;
+
+  /**
+   * Initialize the product module.
+   */
+  public static function init(): void {
+    $obj = new self();
+    if (get_option(self::FIELD_ENFORCE_CAT_LINKS)) {
+      add_filter('post_type_link', [$obj, 'get_product_permalink'], 10, 2);
+    }
+    add_action('admin_init', [$obj, 'register_product_settings']);
+    add_action('admin_init', [$obj, 'save_product_settings']);
```

Comment:
```suggestion
    add_action('admin_init', __CLASS__ . '::register_product_settings');
    add_action('admin_init', __CLASS__ . '::save_product_settings');
```

Reply:
there are many ways to do this, which one is the right one is the question? abusing the static functions without creating a proper object context is not what I'd do 100% of times... I'd really like to discuss it within the team if is also necessary

Reply:
@chrisbaltazar — I agree this is debatable in terms of best practices, but we are talking consistency here. If we decide to change the way we do things from now we should all adopt a new way, but until then we should stick to one standard as much as possible.